pkg/controller/logstorage/elastic/elastic_controller.go

```diff
@@ -0,0 +1,734 @@
+		{Name: render.KibanaPolicyName, Namespace: render.KibanaNamespace},
+		{Name: render.ECKOperatorPolicyName, Namespace: render.ECKOperatorNamespace},
+		{Name: render.ElasticsearchInternalPolicyName, Namespace: render.ElasticsearchNamespace},
+		{Name: networkpolicy.TigeraComponentDefaultDenyPolicyName, Namespace: render.ElasticsearchNamespace},
+		{Name: networkpolicy.TigeraComponentDefaultDenyPolicyName, Namespace: render.KibanaNamespace},
+		{Name: esgateway.PolicyName, Namespace: render.ElasticsearchNamespace},
+		{Name: esmetrics.ElasticsearchMetricsPolicyName, Namespace: render.ElasticsearchNamespace},
+		{Name: kubecontrollers.EsKubeControllerNetworkPolicyName, Namespace: common.CalicoNamespace},
+		{Name: linseed.PolicyName, Namespace: render.ElasticsearchNamespace},
+	})
+
+	// Watch for changes in storage classes, as new storage classes may be made available for LogStorage.
+	err = c.Watch(&source.Kind{Type: &storagev1.StorageClass{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return fmt.Errorf("log-storage-elastic-controller failed to watch StorageClass resource: %w", err)
+	}
+
+	if err = c.Watch(&source.Kind{Type: &apps.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{Namespace: render.ECKOperatorNamespace, Name: render.ECKOperatorName},
+	}}, &handler.EnqueueRequestForObject{}); err != nil {
+		return fmt.Errorf("log-storage-elastic-controller failed to watch StatefulSet resource: %w", err)
+	}
+	if err = c.Watch(&source.Kind{Type: &esv1.Elasticsearch{
+		ObjectMeta: metav1.ObjectMeta{Namespace: render.ElasticsearchNamespace, Name: render.ElasticsearchName},
+	}}, &handler.EnqueueRequestForObject{}); err != nil {
+		return fmt.Errorf("log-storage-elastic-controller failed to watch Elasticsearch resource: %w", err)
+	}
+	if err = c.Watch(&source.Kind{Type: &kbv1.Kibana{
+		ObjectMeta: metav1.ObjectMeta{Namespace: render.KibanaNamespace, Name: render.KibanaName},
+	}}, &handler.EnqueueRequestForObject{}); err != nil {
+		return fmt.Errorf("log-storage-elastic-controller failed to watch Kibana resource: %w", err)
+	}
+
+	// Watch all the elasticsearch user secrets in the operator namespace.
+	if err = utils.AddSecretWatchWithLabel(c, common.OperatorNamespace(), logstoragecommon.TigeraElasticsearchUserSecretLabel); err != nil {
+		return fmt.Errorf("log-storage-elastic-controller failed to watch Secret resource: %w", err)
+	}
+
+	for _, secretName := range []string{
+		render.TigeraElasticsearchGatewaySecret, render.TigeraKibanaCertSecret,
+		render.OIDCSecretName, render.DexObjectName, esmetrics.ElasticsearchMetricsServerTLSSecret,
+		render.TigeraLinseedSecret, certificatemanagement.CASecretName,
+	} {
+		if err = utils.AddSecretsWatch(c, secretName, common.OperatorNamespace()); err != nil {
+			return fmt.Errorf("log-storage-elastic-controller failed to watch Secret resource: %w", err)
+		}
+	}
+	if err = utils.AddSecretsWatch(c, relasticsearch.PublicCertSecret, common.OperatorNamespace()); err != nil {
```

Comment:
I think all the secrets with OperatorNamespace can be added to the for-loop.

Reply:
In general, I think the watches in this func can be refactored into a few for-loops

Reply:
To be honest, I was planning on getting rid of the for loop anyway. This mix/match pre-dates my changes and I'd like to refactor to make it all consistent.